internal/pkg/externalplugins/blunderbuss/blunderbuss.go

```diff
@@ -196,47 +200,111 @@ func HandleIssueCommentEvent(gc githubClient, ce *github.IssueCommentEvent, cfg
 
-	if len(reviewers) > 0 {
-		log.Infof("Requesting reviews from users %s.", reviewers)
-		return ghc.RequestReview(repo.Owner.Login, repo.Name, pr.Number, reviewers)
+	// Filter out unavailable contributors.
+	for contributor := range contributors {
+		if !availableReviewers.Has(contributor) {
+			delete(contributors, contributor)
+		}
 	}
-	return nil
+
+	// The default weight for other reviewers is 1.
+	for _, reviewer := range availableReviewers.List() {
+		_, ok := contributors[reviewer]
+		if !ok {
+			contributors[reviewer] = defaultWeight
+		}
+	}
+	// Create weighted selectors chooser on the number of changes made to the code.
+	var choices []wr.Choice
+	for contributor, weight := range contributors {
+		choices = append(choices, wr.Choice{
+			Item:   contributor,
+			Weight: weight,
+		})
+	}
+	reviewers := sets.NewString()
+
+	chooser, err := wr.NewChooser(
+		choices...,
+	)
+	if err != nil {
+		return err
+	}
+	for len(reviewers) < maxReviewerCount {
+		// Rand pick.
+		reviewers.Insert(chooser.Pick().(string))
+	}
+
+	log.Infof("Requesting reviews from users %s.", reviewers.List())
+	return gc.RequestReview(repo.Owner.Login, repo.Name, pr.Number, reviewers.List())
 }
 
-func getReviewers(author string, reviewers []string, excludeReviewers []string, log *logrus.Entry) []string {
+func listAvailableReviewers(author string, reviewers []string, excludeReviewers []string,
+	requestedReviewers []github.User) sets.String {
 	authorSet := sets.NewString(github.NormLogin(author))
 	excludeReviewersSet := sets.NewString(excludeReviewers...)
+	requestedReviewersSet := sets.NewString()
+	for _, reviewer := range requestedReviewers {
+		requestedReviewersSet.Insert(reviewer.Login)
+	}
 	reviewersSet := sets.NewString()
 	reviewersSet.Insert(reviewers...)
 
-	var result []string
-	// Exclude the author.
-	availableReviewers := layeredsets.NewString(
-		reviewersSet.Difference(authorSet).Difference(excludeReviewersSet).List()...)
+	return reviewersSet.Difference(authorSet).Difference(excludeReviewersSet).Difference(requestedReviewersSet)
+}
 
-	for availableReviewers.Len() > 0 {
-		reviewer := availableReviewers.PopRandom()
-		result = append(result, reviewer)
-		log.Infof("Added %s as reviewers. %d reviewers found.", reviewer, len(result))
+func listChangesContributors(gc githubClient, org string, repo string, num int,
+	log *logrus.Entry) (map[string]uint, error) {
+	changes, err := gc.GetPullRequestChanges(org, repo, num)
+	if err != nil {
+		return nil, fmt.Errorf("error get pull request changes: %v", err)
+	}
+
+	contributors := make(map[string]uint)
+	for _, change := range changes {
+		commits, err := gc.ListFileCommits(org, repo, change.Filename)
+		if err != nil {
+			log.WithError(err).Warnf("Failed list file commits of %s.", change.Filename)
+		}
+
+		for _, commit := range commits {
+			contributor := commit.Author.Login
+			weight, ok := contributors[contributor]
+			if ok {
+				contributors[contributor] = weight + defaultWeight
```

Comment:
Can `defaultWeight` be understood as the increment of weight`?

Reply:
Yes.

Reply:
I feel that the name is not very clear.

Reply:
What your suggestions?

Reply:
what about `weightIncrement`, or `defaultWeightIncrement`?

Reply:
Added.